includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1270,6 +1270,9 @@ public function process_payment( $order_id ) {
 				);
 
 				$error_details = esc_html( rtrim( $e->getMessage(), '.' ) );
+				if ( $e instanceof API_Exception && null !== $e->get_merchant_message() ) {
```

Comment:
`null !== $e->get_merchant_message()` Maybe we can do an empty check here instead, to avoid possible empty merchant messages.

Reply:
I don't think it can be an empty string, and with d5b199989b60281a4d2715c2f3a65c19edd194bd, we do not need to check it anymore. Just a check with type `API_Merchant_Exception` is enough.